scripts.js

```diff
@@ -0,0 +1,205 @@
+                }
+                
+            }
+            
+        }
+     }
+    _addDev(devname){
+        for (let developer of [...staff.developers]){
+            let curNewDev = {
+                name: devname,
+                department: this.department,
+                manager: this.name
+            }
+            for (let managItem of [...staff.managers]){
+                if(this.department === managItem.department){
+                    managItem.developers.push(devname);
+                }
+            }
+            
+            staff.developers.push(curNewDev);
+            return this.developers;
+            
+        }
+     }
+    
+}
+
+class Developer extends Employee {
+    constructor(name, dateOfBirth, salary, department, manager){
+        super(name, dateOfBirth, salary, department);
+        this.manager = manager;
+        this.addToStaff()
+    }
+    addToStaff(){
+        let devItem = {
+            name: this.name,
+            department: this.department,
+            manager: this.manager
+        }
+        staff.developers.push(devItem);
+    }
+    _changeManager(mname){
+        for(let manager of staff.managers){
+            if(manager.name === mname){
+                for(let developer of staff.developers){
+                    if(this.name === developer.name){
+                        this.manager = manager.name;
+                        this.department = manager.department;
+                        developer.manager = this.manager;
+                        developer.department = this.department;
+                    }
+                }
+                manager.developers.push(this.name);
+            } else {
+                let curDevs = manager.developers;
+                manager.developers = curDevs.filter(curDevs => curDevs.includes(this.name) !== true)
+            }
+        }
+     }
+}
+
+let alex = new Human('Alex', '1987, 4, 15');
+alex.displayinfo();//Инфо об Алексе как о человеке
+
+let emp_alex = new Employee('Alex', '1987, 4, 15', 30000, 'IT');
+emp_alex.displayinfo();//Инфо об Алексе как о сотруднике
+
+let m1 = new Manager('John', '1987, 4, 15', 30000, 'Dev', ['Jack', 'Daniel']);
+let m2 = new Manager('Bruce', '1989, 7, 1', 30000, 'Support', ['Monica']);
+let m3 = new Manager('Nancy', '1991, 8, 13', 24000, 'Marketing', ['Max']);
+
+let d1 = new Developer('Jack', '1995, 3, 25', 34000, 'Dev', 'John');
+let d2 = new Developer('Daniel', '1996, 10, 1', 32000, 'Dev', 'John');
+let d3 = new Developer('Monica', '1996, 3, 15', 30000, 'Support', 'Bruce');
+let d4 = new Developer('Max', '1996, 3, 15', 30000, 'Marketing', 'Nancy');
+
+m1._removeDev('Jack');//Убираем Джека из подчинения Джона
+m2._addDev('Helen');//Нанимаем Хелен в подчинение к Брюсу
+d3._changeManager('Nancy')//Переводим Монику в отдел маркетинга к Нэнси
+
+console.log(staff);
+
+
+
+
+
+/*п. 4*/
+
+let promiseArray = [];
+for (let i = 1; i <= 10; i++){
+    promiseArray[i] = new Promise(function(resolve, reject){
```

Comment:
```suggestion
    promiseArray[i] = 
```
fetch() возвращает promise, который дальше обрабатывается  методом then(), который тоже возвращает promise, который обрабатывается следующим then(), который также возвращает promise - именно этот promise помещается в promiseArray[i]